pkg/transformer/eth_getTransactionByHash.go

```diff
@@ -19,109 +20,209 @@ func (p *ProxyETHGetTransactionByHash) Method() string {
 	return "eth_getTransactionByHash"
 }
 
-func (p *ProxyETHGetTransactionByHash) Request(rawreq *eth.JSONRPCRequest) (interface{}, error) {
-	var req eth.GetTransactionByHashRequest
-	if err := unmarshalRequest(rawreq.Params, &req); err != nil {
-		return nil, err
+func (p *ProxyETHGetTransactionByHash) Request(req *eth.JSONRPCRequest) (interface{}, error) {
+	var txHash eth.GetTransactionByHashRequest
+	if err := json.Unmarshal(req.Params, &txHash); err != nil {
+		return nil, errors.Wrap(err, "couldn't unmarshal request")
+	}
+	if txHash == "" {
+		return nil, errors.New("transaction hash is empty")
 	}
 
-	qtumreq := p.ToRequest(&req)
-
-	return p.request(qtumreq)
+	qtumReq := &qtum.GetTransactionRequest{
+		TxID: utils.RemoveHexPrefix(string(txHash)),
+	}
+	return p.request(qtumReq)
 }
 
 func (p *ProxyETHGetTransactionByHash) request(req *qtum.GetTransactionRequest) (*eth.GetTransactionByHashResponse, error) {
-	var tx *qtum.GetTransactionResponse
-	if err := p.Qtum.Request(qtum.MethodGetTransaction, req, &tx); err != nil {
-		if err == qtum.EmptyResponseErr {
-			return nil, nil
-		}
-
+	ethTx, err := getTransactionByHash(p.Qtum, req.TxID)
+	if err != nil {
 		return nil, err
 	}
+	return ethTx, nil
+}
 
-	ethVal, err := QtumAmountToEthValue(tx.Amount)
+// TODO: think of returning flag if it's a rewrad transaction for miner
+func getTransactionByHash(p *qtum.Qtum, hash string) (*eth.GetTransactionByHashResponse, error) {
+	qtumTx, err := p.GetTransaction(hash)
 	if err != nil {
+		// TODO: implement typed error at Qtum pkg
+		// * Filter "transaction not found" error
+		//
+		// TODO: researching
+		// ?! Is the case only for reward transactions
+		cause := errors.Cause(err)
+		if !strings.Contains(cause.Error(), "[code: -5]") {
+			return nil, err
+		}
 
-		return nil, err
+		ethTx, err := getRewardTransactionByHash(p, hash)
+		if err != nil {
+			return nil, errors.WithMessage(err, "couldn't get reward transaction by hash")
+		}
+		return ethTx, nil
+	}
+	qtumDecodedRawTx, err := p.DecodeRawTransaction(qtumTx.Hex)
+	if err != nil {
+		return nil, errors.WithMessage(err, "couldn't get raw transaction")
+	}
+
+	ethTx := &eth.GetTransactionByHashResponse{
+		Hash: utils.AddHexPrefix(qtumDecodedRawTx.ID),
+		// TODO: researching
+		// ? May be don't need this
+		// ! Probably, needs huge quantity requests to calc by hands
+		Nonce: "0x0",
```

Comment:
You don't need this.

Reply:
Okay! In this case, we assume that this value will always be "0x0" to keep the correct value format as the graph-node requires. Issue is resolved, see commit by hash: aa081624c7cd018f21f47eb4688b1adfdb6c500e